diff --git a/node_modules/spotify-types/typings/album.d.ts b/node_modules/spotify-types/typings/album.d.ts
index 6b8ea28..10b6c72 100644
--- a/node_modules/spotify-types/typings/album.d.ts
+++ b/node_modules/spotify-types/typings/album.d.ts
@@ -72,7 +72,7 @@ export interface SimplifiedAlbum {
     /** The total number of tracks in the album. */
     total_tracks: number;
     /** The object type which will be 'album'. */
-    type: SpotifyType;
+    type: Extract<SpotifyType, "album">;
     /** The Spotify URI for the album. */
     uri: string;
 }
\ No newline at end of file
diff --git a/node_modules/spotify-types/typings/artist.d.ts b/node_modules/spotify-types/typings/artist.d.ts
index e8d445d..fb354b4 100644
--- a/node_modules/spotify-types/typings/artist.d.ts
+++ b/node_modules/spotify-types/typings/artist.d.ts
@@ -16,7 +16,7 @@ export interface SimplifiedArtist {
     /** The name of the artist. */
     name: string;
     /** The object type: "artist". */
-    type: SpotifyType;
+    type: Extract<SpotifyType, "artist">;
     /** The Spotify URI for the artist. */
     uri: string;
 }
diff --git a/node_modules/spotify-types/typings/episode.d.ts b/node_modules/spotify-types/typings/episode.d.ts
index 4d4a396..75753b6 100644
--- a/node_modules/spotify-types/typings/episode.d.ts
+++ b/node_modules/spotify-types/typings/episode.d.ts
@@ -59,7 +59,7 @@ export interface SimplifiedEpisode {
     /** The user’s most recent position in the episode. Set if the supplied access token is a user token and has the scope ‘user-read-playback-position’. */
     resume_point?: ResumePoint;
     /** The object type: “episode”. */
-    type: SpotifyType;
+    type: Extract<SpotifyType, "episode">;
     /** The Spotify URI for the episode */
     uri: string;
 }
diff --git a/node_modules/spotify-types/typings/player.d.ts b/node_modules/spotify-types/typings/player.d.ts
index 35f48c0..5bce18a 100644
--- a/node_modules/spotify-types/typings/player.d.ts
+++ b/node_modules/spotify-types/typings/player.d.ts
@@ -195,7 +195,7 @@ export interface RecentlyPlayed {
  */
 export interface PlayHistory {
     /** The context the track was played from. */
-    context: PlayerContext;
+    context: PlayerContext | null;
     /** The date and time the track was played. */
     played_at: string;
     /** The track the user listened to. */
diff --git a/node_modules/spotify-types/typings/playlist.d.ts b/node_modules/spotify-types/typings/playlist.d.ts
index 2a23a39..027cc99 100644
--- a/node_modules/spotify-types/typings/playlist.d.ts
+++ b/node_modules/spotify-types/typings/playlist.d.ts
@@ -60,7 +60,7 @@ export interface SimplifiedPlaylist {
     /** The Spotify URI for the playlist. */
     uri: string;
     /** The object type: “playlist” */
-    type: SpotifyType;
+    type: Extract<SpotifyType, "playlist">;
 }
 
 /**
diff --git a/node_modules/spotify-types/typings/show.d.ts b/node_modules/spotify-types/typings/show.d.ts
index 281a6d5..ca586a6 100644
--- a/node_modules/spotify-types/typings/show.d.ts
+++ b/node_modules/spotify-types/typings/show.d.ts
@@ -51,7 +51,7 @@ export interface SimplifiedShow {
     /** The publisher of the show. */
     publisher: string;
     /** The object type: “show”. */
-    type: SpotifyType;
+    type: Extract<SpotifyType, "show">;
     /** The Spotify URI for the show. */
     uri: string;
 }
\ No newline at end of file
diff --git a/node_modules/spotify-types/typings/track.d.ts b/node_modules/spotify-types/typings/track.d.ts
index a33e0ea..edeb1c2 100644
--- a/node_modules/spotify-types/typings/track.d.ts
+++ b/node_modules/spotify-types/typings/track.d.ts
@@ -44,7 +44,7 @@ export interface SimplifiedTrack {
     /** The number of the track. If an album has several discs, the track number is the number on the specified disc. */
     track_number: number;
     /** The object type: “track”. */
-    type: SpotifyType;
+    type: Extract<SpotifyType, "track">;
     /** The Spotify URI for the track. */
     uri: string;
 }
diff --git a/node_modules/spotify-types/typings/user.d.ts b/node_modules/spotify-types/typings/user.d.ts
index 6267be5..8dd0b4d 100644
--- a/node_modules/spotify-types/typings/user.d.ts
+++ b/node_modules/spotify-types/typings/user.d.ts
@@ -69,7 +69,7 @@ export interface PublicUser {
     /** The Spotify URI for the user. */
     uri: string;
     /** The Spotify object type which will be 'User'. */
-    type: SpotifyType;
+    type: Extract<SpotifyType, "user">;
     /** The user’s profile image. */
     images?: Image[];
     /** Information about the followers of the user. */
